contrib/aero-snap/main.cpp

```
@@ -0,0 +1,560 @@
+    void rememberRestoreBox(const PHLWINDOW& window, const CBox& box) {
+        if (recordFor(window))
+            return;
+
+        state->records.push_back({window, box});
+    }
+
+    void forgetRestoreBox(const PHLWINDOW& window) {
+        std::erase_if(state->records, [&](const auto& record) { return record.window.lock() == window; });
+    }
+
+    bool floatingModeActive() {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)